### PR TITLE
ci: include libyaml in image and use docker compose

### PR DESCRIPTION
**Pull request by TheShubhendra**
This pull request includes updates to the CI workflow to address build issues.

### Changes
1. **Build libyaml before installing lyaml**:
   - Added a step to build and install libyaml before attempting to install the lyaml Lua module.
   - This change resolves the issue where lyaml installation fails due to missing libyaml symbols.

2. **Switch from docker-compose to docker compose**:
   - Updated the CI workflow to replace `docker-compose` with `docker compose` for compatibility with newer Docker versions.

### CODE REVIEW

File: debian-dev/Dockerfile.local [hunk 1]
```diff
@@ -35,6 +35,15 @@ RUN set -x \
         make \
         git  \
         sudo \
+        clang \
+        wget \
+    && wget https://github.com/yaml/libyaml/releases/download/0.2.5/yaml-0.2.5.tar.gz \
```
Comment (shreemaan-abhishek) [inline debian-dev/Dockerfile.local:40]: instead, apt install libyaml-dev would suffice isn't it?

Comment (TheShubhendra) [reply debian-dev/Dockerfile.local:40]: previously I had did that only, but when luarocks installs lyaml, it was not able to find  C symbols from those libyaml.so files.

Comment (TheShubhendra) [reply debian-dev/Dockerfile.local:40]: I was getting the same error as mentioned here https://github.com/gvvaughan/lyaml/issues/30 .

**Comment by TheShubhendra** [comment]
@shreemaan-abhishek Can you please run the workflows?

**Comment by kayx23** [comment]
all checks passed

**Review by shreemaan-abhishek — APPROVED** [review]
LGTM

**Review by moonming — APPROVED** [review]

**Review by shreemaan-abhishek — APPROVED** [review]

**Pull request merged**